helpers/analysis.js

```diff
@@ -0,0 +1,159 @@
+  array.forEach(function(element) {
+    var res = element.result.name.split("=");
+    if (i.name === res[res.length - 1]) {
+      // console.log(i.name);
+      array.push({
+        result: {
+          name: element.result.name + "=" + j.name,
+          latitude: element.result.latitude + "=" + j.latitude,
+          longitude: element.result.longitude + "=" + j.longitude,
+          address1: element.result.address1 + "=" + j.address1,
+          address2: element.result.address2 + "=" + j.address2
+        }
+      });
+    }
+  });
+  array.push({
+    result: {
+      name: i.name + "=" + j.name,
+      latitude: i.latitude + "=" + j.latitude,
+      longitude: i.longitude + "=" + j.longitude,
+      address1: i.address1 + "=" + j.address1,
+      address2: i.address2 + "=" + j.address2
+    }
+  });
+};
+
+
+// analyze all search results to find locations which meet conditions
+var analyzeResult = function(displayedArray) {
+  var analyzeArray0 = [];
+  var analyzeArray1 = [];
+  var analyzeArray2 = [];
+  var analyzeArray3 = [];
+  var displayAnalyzeObject = {
+    tier0: [],
+    tier1: [],
+    tier2: [],
+    tier3: []
+  };
+
+  for (var index = 0; index < displayedArray.length; index++) {
+    // iterate through 2 adjacent arrays
+    displayedArray[index].indivResult.forEach(function(i) {
+      // console.log("test1");
+      if (index + 1 < displayedArray.length) {
+        displayedArray[index + 1].indivResult.forEach(function(j) {
+          // console.log("test2");
+          // var analyzeString = analyzeResult(i, j);
+
+          // check for distance
+          switch (analyzeDistance(i, j)) {
+            case 0:
+              joinLocation(analyzeArray0, i, j);
+              break;
+            case 1:
+              joinLocation(analyzeArray1, i, j);
+              break;
+            case 2:
+              joinLocation(analyzeArray2, i, j);
+              break;
+            case 3:
+              joinLocation(analyzeArray3, i, j);
+              break;
+          }
+        });
+      }
+    });
+  }
+
+  // check for elements which are present for all search fields
+  // console.log(analyzeArray);
+  analyzeArray0.forEach(function(element) {
+    var res = element.result.name.split("=");
+    if (res.length === displayedArray.length) {
+      displayAnalyzeObject.tier0.push(element);
+    }
+  });
+  analyzeArray1.forEach(function(element) {
+    var res = element.result.name.split("=");
+    if (res.length === displayedArray.length) {
+      displayAnalyzeObject.tier1.push(element);
+    }
+  });
+  analyzeArray2.forEach(function(element) {
+    var res = element.result.name.split("=");
+    if (res.length === displayedArray.length) {
+      displayAnalyzeObject.tier2.push(element);
+    }
+  });
+  analyzeArray3.forEach(function(element) {
+    var res = element.result.name.split("=");
+    if (res.length === displayedArray.length) {
+      displayAnalyzeObject.tier3.push(element);
+    }
+  });
+
```

Comment:
a lot of loops happened here, please explain it to me when you meet me why this decisions are taken.